### PR TITLE
Add feature to mount host volumes

### LINKED ISSUE #475: Support mounting volumes for install/uninstall commands
I have a use case in which I would like to bind-mount a file in the install process.
After exploring the codebase, it is not clear whether it is possible or not to define volumes/mount points in the porter.yml or in any other place.

Is this something left to mixins specification?

### PR DESCRIPTION
**Pull request by maniSbindra**
# What does this change
This change adds flag **--mount-host-volume** for commands like **porter install** and **porter uninstall**, which enables making a host file system folder available inside the running container, which executes the action.

For example the following commands makes the host's **$HOME/.hostfolder** folder available inside the bundle (as **read/write** mount) at **/home/nonroot/.hostfolder** .

```
porter install test-install --mount-host-volume "$HOME/.hostfolder:/home/nonroot/.hostfolder:rw"
```

Multiple folders may be mounted. Format for adding a mount is **\<host path\>:\<container path\>\[:\<option\>\]**. If option is **ro** or not provided or invalid, then file system is mounted as **read only**. If option is **rw** then file system is mounted as **read/write** inside the container

# What issue does it fix
Closes #475 

# Notes for the reviewer
This works with the docker driver. By default folders are mounted as read only.

# Checklist
- [X] Did you write tests?
- [ ] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

# Reviewer Checklist
* Comment with /azp run test-porter-release if a magefile or build script was modified
* Comment with /azp run porter-integration if it's a non-trivial PR

[contributors]: https://getporter.org/src/CONTRIBUTORS.md

### CODE REVIEW

**Comment by lbergnehr** [comment]
This is great! I would suggest, if it's not already there, to also enable mounting through the porter config. That way you vould set up mounts you always want, for example `~/.azure` when doing Azure CLI authentication. Would be even more seamless.

File: pkg/cnab/provider/docker_linux.go [hunk 1]
```diff
@@ -47,3 +47,14 @@ func (r *Runtime) mountDockerSocket(cfg *container.Config, hostCfg *container.Ho
 	hostCfg.Mounts = append(hostCfg.Mounts, dockerSockMount)
 	return nil
 }
+
+func (r *Runtime) addVolumeMountToHostConfig(hostConfig *container.HostConfig, source string, target string, readOnly bool) error {
+	mount := mount.Mount{
+		Source:   source,
+		Target:   target,
+		Type:     "bind",
```
Comment (sgettys) [inline pkg/cnab/provider/docker_linux.go:55]: Should this be volume instead of bind?

File: pkg/porter/lifecycle.go [hunk 2]
```diff
@@ -92,6 +99,47 @@ func (o *BundleExecutionOptions) GetParameters() map[string]interface{} {
 	return o.finalParams
 }
 
+// Sets the final resolved set of host volumes to be made availabe to the bundle
+func (o *BundleExecutionOptions) GetHostVolumeMounts() []cnabprovider.HostVolumeMountSpec {
```
Comment (sgettys) [inline pkg/porter/lifecycle.go:103]: Will this handle spaces in the windows directory structure?

File: pkg/cnab/provider/docker_linux.go (hunk 1 above)
Comment (maniSbindra) [reply pkg/cnab/provider/docker_linux.go:55]: I did try with volume, however then you cannot mount a target folder where the name of the folder starts with a dot, the error received is something like `cannot create container: Error response from daemon: create /abc/.xyz: "/Users/testuser/.xyx" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed`

File: pkg/porter/lifecycle.go (hunk 2 above)
Comment (maniSbindra) [reply pkg/porter/lifecycle.go:103]: Yes it does work. I have now added a test for the same. Thanks

**Review by sgettys — APPROVED** [review]

**Pull request merged**